src/bundle/DependencyInjection/SchebTwoFactorExtension.php

```diff
@@ -67,6 +69,13 @@ public function load(array $configs, ContainerBuilder $container): void
         }
 
         $this->configureBackupCodeManager($container, $config);
+
+        if (isset($config['rate_limiter'])) {
```

Comment:
This should check if the rate-limiter component is present and if not throw a meaningful error message to let the user know, they have to install that components (to make it optional and only require people to install it, when it's actually needed).

Reply:
I can add a check for the component too. Does that mean you want to remove `symfony/rate-limiter` from the dev dependencies? We need to check if the key is set as well, because we need it to inject the right rate limiter service.

Reply:
Resolved via https://github.com/scheb/2fa/pull/165#discussion_r1038863299, no need to change.